scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -149,6 +149,11 @@ public static BuildTargetResult RestoreLockedCoreHost(BuildTargetContext c)
             return c.Success();
         }
 
+        [Target(
+            nameof(CompileCoreHost),
+            nameof(BuildCoreHostPackages))]
+        public static BuildTargetResult BuildCoreHost(BuildTargetContext c) => c.Success();
```

Comment:
Ah, I get your point now.

Yes, having VSO know about the internals of our build is not a good thing going forward.  :(

This change will actually clean VSO up a little bit.  Instead of saying:

```
build -Configuration $(BuildConfiguration) -Targets Prepare,CompileCoreHost
Sign
build -Configuration $(BuildConfiguration) -Targets Prepare,PackagePkgProjects,RestoreLockedCoreHost,...
```

It can now say:

```
build -Configuration $(BuildConfiguration) -Targets Prepare,CompileCoreHost
Sign
build -Configuration $(BuildConfiguration) -Targets Prepare,BuildCoreHostPackages,...
```

Either way I'm going to have to change VSO since I'm injecting a new step between `PackagePkgProjects` and `RestoreLockedCoreHost`